chapters/2-abstraction.md

```diff
@@ -0,0 +1,100 @@
+# Abstraction
+
+Functional abstraction allows us to treat functions as black boxes.
+We just care about what a function does and how to use it.
```

Comment:
I think the value of this paragraph is that it caters to the clueless students. The non-blur ones will get stuck on the application, not the definition here. Basically need to help them think.

1) Need to say what we do not care about. Else, students might say "okay. so we need to care about what a function does and how to use it. Isn't that everything?" "What a function does? So does the function process the values using a for loop or while loop?"

2) Maybe make the logical link explicit. "... boxes, meaning that, we just care ...". The clueless ones might not realize that you are explaining what you mean by "black boxes".